git-cliff-1.4.0/man/git-cliff.1
.ie \n(.g .ds Aq \(aq
.el .ds Aq '
.TH git-cliff 1  "git-cliff 1.4.0" 
.SH NAME
git\-cliff \- A highly customizable changelog generator ⛰️
.SH SYNOPSIS
\fBgit\-cliff\fR [\fB\-h\fR|\fB\-\-help\fR] [\fB\-V\fR|\fB\-\-version\fR] [\fB\-v\fR|\fB\-\-verbose\fR]... [\fB\-c\fR|\fB\-\-config\fR] [\fB\-w\fR|\fB\-\-workdir\fR] [\fB\-r\fR|\fB\-\-repository\fR] [\fB\-\-include\-path\fR] [\fB\-\-exclude\-path\fR] [\fB\-\-with\-commit\fR] [\fB\-p\fR|\fB\-\-prepend\fR] [\fB\-o\fR|\fB\-\-output\fR] [\fB\-t\fR|\fB\-\-tag\fR] [\fB\-\-bump\fR] [\fB\-b\fR|\fB\-\-body\fR] [\fB\-i\fR|\fB\-\-init\fR] [\fB\-l\fR|\fB\-\-latest\fR] [\fB\-\-current\fR] [\fB\-u\fR|\fB\-\-unreleased\fR] [\fB\-\-topo\-order\fR] [\fB\-\-context\fR] [\fB\-s\fR|\fB\-\-strip\fR] [\fB\-\-sort\fR] [\fIRANGE\fR] 
.SH DESCRIPTION
A highly customizable changelog generator ⛰️
.SH OPTIONS
.TP
\fB\-h\fR, \fB\-\-help\fR
Prints help information
.TP
\fB\-V\fR, \fB\-\-version\fR
Prints version information
.TP
\fB\-v\fR, \fB\-\-verbose\fR
Increases the logging verbosity
.TP
\fB\-c\fR, \fB\-\-config\fR=\fIPATH\fR [default: cliff.toml]
Sets the configuration file
.RS
May also be specified with the \fBGIT_CLIFF_CONFIG\fR environment variable. 
.RE
.TP
\fB\-w\fR, \fB\-\-workdir\fR=\fIPATH\fR
Sets the working directory
.RS
May also be specified with the \fBGIT_CLIFF_WORKDIR\fR environment variable. 
.RE
.TP
\fB\-r\fR, \fB\-\-repository\fR=\fIPATH\fR
Sets the git repository
.RS
May also be specified with the \fBGIT_CLIFF_REPOSITORY\fR environment variable. 
.RE
.TP
\fB\-\-include\-path\fR=\fIPATTERN\fR
Sets the path to include related commits
.RS
May also be specified with the \fBGIT_CLIFF_INCLUDE_PATH\fR environment variable. 
.RE
.TP
\fB\-\-exclude\-path\fR=\fIPATTERN\fR
Sets the path to exclude related commits
.RS
May also be specified with the \fBGIT_CLIFF_EXCLUDE_PATH\fR environment variable. 
.RE
.TP
\fB\-\-with\-commit\fR=\fIMSG\fR
Sets custom commit messages to include in the changelog
.RS
May also be specified with the \fBGIT_CLIFF_WITH_COMMIT\fR environment variable. 
.RE
.TP
\fB\-p\fR, \fB\-\-prepend\fR=\fIPATH\fR
Prepends entries to the given changelog file
.RS
May also be specified with the \fBGIT_CLIFF_PREPEND\fR environment variable. 
.RE
.TP
\fB\-o\fR, \fB\-\-output\fR=\fIPATH\fR
Writes output to the given file
.RS
May also be specified with the \fBGIT_CLIFF_OUTPUT\fR environment variable. 
.RE
.TP
\fB\-t\fR, \fB\-\-tag\fR=\fITAG\fR
Sets the tag for the latest version
.RS
May also be specified with the \fBGIT_CLIFF_TAG\fR environment variable. 
.RE
.TP
\fB\-\-bump\fR
Bumps the version for unreleased changes
.TP
\fB\-b\fR, \fB\-\-body\fR=\fITEMPLATE\fR
Sets the template for the changelog body
.RS
May also be specified with the \fBGIT_CLIFF_TEMPLATE\fR environment variable. 
.RE
.TP
\fB\-i\fR, \fB\-\-init\fR
Writes the default configuration file to cliff.toml
.TP
\fB\-l\fR, \fB\-\-latest\fR
Processes the commits starting from the latest tag
.TP
\fB\-\-current\fR
Processes the commits that belong to the current tag
.TP
\fB\-u\fR, \fB\-\-unreleased\fR
Processes the commits that do not belong to a tag
.TP
\fB\-\-topo\-order\fR
Sorts the tags topologically
.TP
\fB\-\-context\fR
Prints changelog context as JSON
.TP
\fB\-s\fR, \fB\-\-strip\fR=\fIPART\fR
Strips the given parts from the changelog
.br

.br
[\fIpossible values: \fRheader, footer, all]
.TP
\fB\-\-sort\fR=\fISORT\fR [default: oldest]
Sets sorting of the commits inside sections
.br

.br
[\fIpossible values: \fRoldest, newest]
.TP
[\fIRANGE\fR]
Sets the commit range to process
.SH VERSION
v1.4.0
.SH AUTHORS
git\-cliff contributors <git\-cliff@protonmail.com>


git-cliff-1.4.0/completions/git-cliff.bash
_git-cliff() {
    local i cur prev opts cmd
    COMPREPLY=()
    cur="${COMP_WORDS[COMP_CWORD]}"
    prev="${COMP_WORDS[COMP_CWORD-1]}"
    cmd=""
    opts=""

    for i in ${COMP_WORDS[@]}
    do
        case "${cmd},${i}" in
            ",$1")
                cmd="git__cliff"
                ;;
            *)
                ;;
        esac
    done

    case "${cmd}" in
        git__cliff)
            opts="-h -V -v -c -w -r -p -o -t -b -i -l -u -s --help --version --verbose --config --workdir --repository --include-path --exclude-path --with-commit --prepend --output --tag --bump --body --init --latest --current --unreleased --topo-order --context --strip --sort [RANGE]"
            if [[ ${cur} == -* || ${COMP_CWORD} -eq 1 ]] ; then
                COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
                return 0
            fi
            case "${prev}" in
                --config)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -c)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --workdir)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -w)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --repository)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -r)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --include-path)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --exclude-path)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --with-commit)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --prepend)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -p)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --output)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -o)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --tag)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -t)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --body)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                -b)
                    COMPREPLY=($(compgen -f "${cur}"))
                    return 0
                    ;;
                --strip)
                    COMPREPLY=($(compgen -W "header footer all" -- "${cur}"))
                    return 0
                    ;;
                -s)
                    COMPREPLY=($(compgen -W "header footer all" -- "${cur}"))
                    return 0
                    ;;
                --sort)
                    COMPREPLY=($(compgen -W "oldest newest" -- "${cur}"))
                    return 0
                    ;;
                *)
                    COMPREPLY=()
                    ;;
            esac
            COMPREPLY=( $(compgen -W "${opts}" -- "${cur}") )
            return 0
            ;;
    esac
}

complete -F _git-cliff -o nosort -o bashdefault -o default git-cliff
